src/Raven.Server/Documents/ShardedHandlers/ShardedSubscriptionsHandler.cs

```diff
@@ -0,0 +1,457 @@
+            }
+        }
+
+        //TODO: egor attribute
+        [RavenShardedAction("/databases/*/subscriptions/connection-details", "GET", AuthorizationStatus.ValidUser, EndpointType.Read/*, CorsMode = CorsMode.Cluster*/)]
+        public async Task GetSubscriptionConnectionDetails()
+        {
+            var subscriptionName = GetStringQueryString("name", false);
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                if (string.IsNullOrEmpty(subscriptionName))
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    return;
+                }
+
+                var state = ShardedContext.ShardedSubscriptionStorage.GetSubscriptionConnection(context, subscriptionName);
+
+                var subscriptionConnectionDetails = new SubscriptionConnectionDetails
+                {
+                    ClientUri = state?.Connection?.ClientUri,
+                    Strategy = state?.Connection?.Strategy
+                };
+
+                context.Write(writer, subscriptionConnectionDetails.ToJson());
+            }
+        }
+
+        //TODO: egor attribute
+        [RavenShardedAction("/databases/*/subscriptions/try", "POST", AuthorizationStatus.ValidUser, EndpointType.Write/*, DisableOnCpuCreditsExhaustion = true*/)]
+        public async Task Try()
+        {
+            using (ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                using var json = await context.ReadForMemoryAsync(RequestBodyStream(), null);
+                var tryout = JsonDeserializationServer.SubscriptionTryout(json);
+                var sub = SubscriptionConnectionBase.ParseSubscriptionQuery(tryout.Query);
+                if (sub.Collection == null)
+                    throw new ArgumentException("Collection must be specified");
+
+                const int maxPageSize = 1024;
+                var pageSize = GetIntValueQueryString("pageSize") ?? 1;
+                if (pageSize > maxPageSize)
+                    throw new ArgumentException($"Cannot gather more than {maxPageSize} results during tryouts, but requested number was {pageSize}.");
+
+                var disposables = new List<IDisposable>();
+                try
+                {
+                    var cmds = new List<SubscriptionTryoutCommand>();
+                    var tasks = new List<Task>();
+                    foreach (var re in ShardedContext.RequestExecutors)
+                    {
+                        disposables.Add(ContextPool.AllocateOperationContext(out TransactionOperationContext ctx));
+                        var cmd = new SubscriptionTryoutCommand(tryout, pageSize);
```

Comment:
I think you also have `timeLimit` which you aren't passing here

Reply:
done